File: rust/2018/src/bin/day_01.rs
use std::collections::HashSet;
use std::fs;

fn main() {
    let text = fs::read_to_string("../input/input_01.txt").unwrap();
    let mut freqs = HashSet::new();
    let changes: Vec<i32> = text.lines().map(|s| s.parse().unwrap()).collect();
    println!("Part 1: {}", changes.iter().sum::<i32>());
    let mut freq = 0;
    for c in changes.iter().cycle() {
        if freqs.contains(&freq) {
            println!("Part 2: {}", freq);
            break;
        }
        freqs.insert(freq);
        freq += c;
    }
}

File: kotlin/y2022/Day01.kt
package y2022

import java.io.File

fun main() {
    val text = File("input/2022/input_01.txt").readText().trim()
    val counts = text.split("\n\n")
        .map { it.lines().sumOf(String::toUInt) }
        .sortedDescending()
    println("Part 1: ${counts[0]}")
    println("Part 2: ${counts.take(3).sum()}")
}
